nickyspatial/stats/spectral.py

```diff
@@ -108,3 +111,232 @@ def attach_spectral_indices(layer, bands=None):
             }
 
     return indices
+
+
+## new --
+class SpectralIndexCalculator:
```

Comment:
Nice creativity man ! I like how this you have defined the indices, however for the custom indices we need proper validation as they can break based on the user input and user needs to know why !